sdk/core/allocator/main.cc

```diff
@@ -1183,7 +1192,7 @@ __cheriot_minimum_stack(0x280) SObj
 			return sealed;
 		}
 	}
-	heap_free(heapCapability, obj);
+	(void)heap_free(heapCapability, obj);
```

Comment:
Would be nice to have a comment here.  This can happen only if the caller gives us a malicious output value.  Actually, we probably can elide this and return sealed unconditionally.  If you give us a valid `unsealed` pointer and we don't write through it, you can still free with the sealed pointer.

Reply:
Moved work to #401.